### PR TITLE
add internal java sol to sleepy cow herding bronze 

### PR DESCRIPTION
**Pull request by sathu-beep**
@thecodingwizard

### CODE REVIEW

File: solutions/USACO Feb19-SleepyCowHerding.mdx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		if (b > c) {
+			temp = b;
+			b = c;
+			c = temp;
+		}
+		if (a > b) {
+			temp = a;
+			a = b;
+			b = temp;
+
```
Comment (maggie-j-liu) [inline solutions/USACO Feb19-SleepyCowHerding.mdx:39]:
```suggestion
```

File: solutions/USACO Feb19-SleepyCowHerding.mdx [hunk 2]
```diff
@@ -0,0 +1,59 @@
+---
+id: usaco-915
```
Comment (maggie-j-liu) [inline solutions/USACO Feb19-SleepyCowHerding.mdx:2]: change the name of the file so that it's the same as the id; in this case, usaco-915.mdx

Comment (sathu-beep) [reply solutions/USACO Feb19-SleepyCowHerding.mdx]: @thecodingwizard sorry about this, this is the first time I'm hearing about this new naming convention

**Review by maggie-j-liu — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]

File: solutions/usaco-915.mdx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+			c = temp;
+		}
+		if (a > b) {
+			temp = a;
+			a = b;
+			b = temp;
+		}
+		if (c == a + 2) {
+			pw.println(0); 
+			pw.print(a)
```
Comment (thecodingwizard) [inline solutions/usaco-915.mdx:42]: this line shouldn't be here (?)

File: solutions/usaco-915.mdx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+		if (b > c) {
+			temp = b;
+			b = c;
+			c = temp;
+		}
+		if (a > b) {
+			temp = a;
+			a = b;
+			b = temp;
+		}
```
Comment (thecodingwizard) [inline solutions/usaco-915.mdx:39]: minor detail, but can we:

- add a comment to the top of this block that says all we're doing is arranging a b c to be in ascending order
- add a comment that all the temp = a / a = b / b = temp code does is swap the variables a and b
- add a blank line around this block to better separate this section from the rest of the code

(unless you think some of these are unnecessary?)

**Review by thecodingwizard — CHANGES_REQUESTED** [review]

File: solutions/usaco-915.mdx (hunk 3 above)
Comment (sathu-beep) [reply solutions/usaco-915.mdx]: rip there goes my streak

Comment (sathu-beep) [reply solutions/usaco-915.mdx]: fixed*

File: solutions/usaco-915.mdx (hunk 4 above)
Comment (sathu-beep) [reply solutions/usaco-915.mdx:43]: minor detail based on preference but i have added these changes

File: solutions/usaco-915.mdx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	public static void main(String[] args) throws IOException {
+		Scanner sc = new Scanner(new File("herding.in"));
+		PrintWriter pw = new PrintWriter(new File("herding.out")); 
+		int a = sc.nextInt();
+		int b = sc.nextInt();
+		int c = sc.nextInt();
+		
+		// we are arranging a, b, and c to be in arranging order
+		// the variable temp helps us swap
+		// a with b, b with c, and c with a
```
Comment (thecodingwizard) [inline solutions/usaco-915.mdx:27]:
```suggestion
		// a with b, b with c, and c with a as needed
```

File: solutions/usaco-915.mdx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+import java.io.*;
+class Main {
+	public static void main(String[] args) throws IOException {
+		Scanner sc = new Scanner(new File("herding.in"));
+		PrintWriter pw = new PrintWriter(new File("herding.out")); 
+		int a = sc.nextInt();
+		int b = sc.nextInt();
+		int c = sc.nextInt();
+		
+		// we are arranging a, b, and c to be in arranging order
```
Comment (maggie-j-liu) [inline solutions/usaco-915.mdx:25]:
```suggestion
		// we are arranging a, b, and c to be in ascending order
```

File: solutions/usaco-915.mdx (hunk 4 above)
Comment (dongliu0426) [reply solutions/usaco-915.mdx:43]: it would be nice to make a swap function instead

**Comment by sathu-beep** [comment]
@thecodingwizard @bqi343 accept orz

**Comment by sathu-beep** [comment]
> sathu added another sol

ye it didn't let me make a new pr with one being active and I didn't want to waste time just waiting so I went ahead and made it in the same pr

**Review by dongliu0426 — CHANGES_REQUESTED** [review]
2nd sol looks fine

**Comment by sathu-beep** [comment]
> 2nd sol looks fine

if its fine then why request changes 🤔

**Comment by maggie-j-liu** [comment]
> > 2nd sol looks fine
> 
> if its fine then why request changes 🤔

make a swap function https://github.com/cpinitiative/usaco-guide/pull/1542#discussion_r634823648

**Review by dongliu0426 — APPROVED** [review]

**Comment by thecodingwizard** [comment]
wait if the swap function is returning an array wouldn't it just make more sense to sort a length 3 array instead?

**Comment by maggie-j-liu** [comment]
i thought @dongliu0426 meant a swap function as in a function that only swapped 2 numbers, but sorting an array is also fine... i don't think it makes a lot of sense to just put the repetitive code inside a function

**Comment by dongliu0426** [comment]
> i thought @dongliu0426 meant a swap function as in a function that only swapped 2 numbers, but sorting an array is also fine... i don't think it makes a lot of sense to just put the repetitive code inside a function

oops that is what i meant.

**Comment by maggie-j-liu** [comment]
I don't think this pr was ready to merge; could you open a new pr to address the requested changes?

**Pull request merged**